### PR TITLE
Implement presence & stats packets

### PR DESCRIPTION
**Pull request by ascenttree**
This pr introduces packet 6 (presence) and packet 5 (stats), and sends them at login. There are still a few fields I'm unsure of, but we'll get there lol

Note for hosters: you will get an error saying "unable to recreate image from data" at login if you don't have an avatar server (http://hexis.dienay.net/a/id).

### CODE REVIEW

File: common/io.go [hunk 1]
```diff
@@ -137,6 +137,22 @@ func (stream *IOStream) ReadString() string {
 	return string(chars)
 }
 
+func (stream *IOStream) ReadIntList() []uint32 {
```
Comment (Lekuruu) [inline common/io.go:140]: Can you also add a `WriteIntList()` function for consistency?

File: hnet/handler.go [hunk 2]
```diff
@@ -34,6 +34,47 @@ func handleLogin(stream *common.IOStream, player *Player) error {
...
+	// fake data for now
+	stats := UserStats{
+		UserId:   player.Id,
+		Rank:     1,
+		Score:    300,
+		Unknown:  1,
+		Unknown2: 2,
+		Accuracy: 0.9914,
+		Plays:    21,
+	}
```
Comment (Lekuruu) [inline hnet/handler.go:51]: Can you attach the presence & stats to the `Player` struct?

**Review by Lekuruu — CHANGES_REQUESTED** [review]

**Comment by Lekuruu** [comment]
I've changed up the player presence & stats a bit, what do you think?

**Comment by Lekuruu** [comment]
I am thinking of renaming the presence to `Info`, because `player.Info.Id` feels better, but I am not sure if that actually makes sense.

**Comment by ascenttree** [comment]
Nothing in the game refers to it as presence, I just named it that because osu! does something similar.
We could rename it to Info, but it's up to you.
Actually, I think the game does call it PlayerInfo aswell

**Comment by Lekuruu** [comment]
Let's see if I can come up with something..

**Comment by Lekuruu** [comment]
I think that works.

**Comment by Lekuruu** [comment]
Do you have a theory on what the `Unknown` stuff could be?

**Comment by ascenttree** [comment]
0 clue
Ghidra sets a local variable equal to it and then seems to do nothing with it so I have 0 idea

**Review by Lekuruu — APPROVED** [review]

**Pull request merged**